thehive/app/org/thp/thehive/services/ShareSrv.scala

```diff
@@ -90,19 +91,44 @@ class ShareSrv @Inject() (
 //    caseSrv.get(`case`).in[ShareCase].filter(_.in[OrganisationShare])._id.getOrFail().flatMap(remove(_)) // FIXME add organisation ?
 
   def remove(shareId: EntityIdOrName)(implicit graph: Graph, authContext: AuthContext): Try[Unit] =
+    unshareCase(shareId).flatMap(_ => Try(get(shareId).remove()))
+
+  def cascadeRemove(shareId: EntityIdOrName)(implicit graph: Graph, authContext: AuthContext): Try[Unit] = {
+    for {
+      organisation <- organisationSrv.getOrFail(authContext.organisation)
+      share        <- getOrFail(shareId)
+
+      tasks          <- Try(get(share).tasks)
```

Comment:
`Try()` are pointlesss. If the method doesn't return a Try use: `tasks = get(share).tasks`

Reply:
Not wrapping in `Try` result in an error : Traversal[Nothing, Nothing, Nothing] is found but Try[Unit] is expected